code/modules/integrated_electronics/subtypes/lists.dm

```diff
@@ -100,6 +101,91 @@
 		activate_pin(3)
 
 
+/obj/item/integrated_circuit/lists/filter
+	name = "filter circuit"
+	desc = "This circuit will search through a list for anything matching the desired element(s) and outputs two lists: \
+	one containing just matching elements, and one with matching elements filtered out."
+	extended_desc = "Sample accepts lists. If no match is found, original list is sent to output 1."
+	inputs = list(
+		"input list" = IC_PINTYPE_LIST,
+		"sample" = IC_PINTYPE_ANY
+		)
+	outputs = list(
+		"list filtered" = IC_PINTYPE_LIST,
+		"list matched" = IC_PINTYPE_LIST
+		)
+	activators = list(
+		"compute" = IC_PINTYPE_PULSE_IN,
+		"on match" = IC_PINTYPE_PULSE_OUT,
+		"on no match" = IC_PINTYPE_PULSE_OUT
+		)
+	complexity = 6
+	icon_state = "addition"
+	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
+
+/obj/item/integrated_circuit/lists/filter/do_work()
+	var/list/input_list = get_pin_data(IC_INPUT, 1)
+	var/sample = get_pin_data(IC_INPUT, 2)
+	var/list/sample_list = islist(sample) ? uniqueList(sample) : null
+	var/list/output_list1 = input_list.Copy()
+	var/list/output_list2 = list()
+	var/list/output = list()
+
+	for(var/i1 in input_list)
```

Comment:
Please use more descriptive variable names for nested loop counters like these. `i`, `j` & `k` are more or less standard (`i` for the outermost loop, then `i` for the one inside it, and then `k` for the one inside `j`) and I will accept them, though I recommend something a bit more verbose like `outer_item` and `inner_item` or something so that you can easily tell at a glance which item is referenced when you mix both inner and outer item references in the same line.

Reply:
Done.